test/functional/multiwallet.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the wallet."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class MultiWalletTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [['-wallet=w1', '-wallet=w2','-wallet=w3']]
+
+    def setup_network(self):
+        self.nodes = self.start_nodes(1, self.options.tmpdir, self.extra_args[:1])
+
+    def run_test(self):
+        self.nodes[0].generate(1)
+
+        #check default wallet balance
+        walletinfo = self.nodes[0].getwalletinfo()
+        assert_equal(walletinfo['immature_balance'], 50)
+
+        #check w1 wallet balance
+        w1 = self.nodes[0] / "/v1/wallet/w1"
+        walletinfo = w1.getwalletinfo()
+        assert_equal(walletinfo['immature_balance'], 50)
+
+        #check w1 wallet balance
+        w2 = self.nodes[0] / "/v1/wallet/w2"
+        walletinfo = w2.getwalletinfo()
+        assert_equal(walletinfo['immature_balance'], 0)
+
+        w3 = self.nodes[0] / "/v1/wallet/w2"
+        walletinfo = w2.getwalletinfo()
```

Comment:
the `w3` assignment is wrong, and this line doesn't appear to be testing anything